sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -1034,6 +1034,10 @@ private void ThrowLockLostException()
             {
                 DateTime[] lockedUntilUtcTimes = amqpResponseMessage.GetValue<DateTime[]>(ManagementConstants.Properties.Expirations);
                 lockedUntil = lockedUntilUtcTimes[0];
+                if (_requestResponseLockedMessages.Contains(lockToken))
```

Comment:
Ran into this while adding new tests - we were not updating the receive deferred lockedUntil time so completing these messages would not work properly if the initial lock time was passed (they would get removed from requestResponseLockedMessages ConcurrentExpiringSet).